includes/abstracts/abstract-wc-data.php

```diff
@@ -445,7 +445,7 @@ public function update_meta_data( $key, $value, $meta_id = 0 ) {
 			}
 		}
 
-		if ( $array_key ) {
+		if ( $array_key !== false ) {
```

Comment:
[This should use a yoda condition](https://travis-ci.org/woocommerce/woocommerce/jobs/445539921#L956) (`false !== $array_key`)

Reply:
I have updated it, Eagerly waiting to see the changes live as this bug affecting our woocommerce extension.